### PR TITLE
Refactor - HTTPServerResponse.end() methods

### PR DESCRIPTION
**Pull request by nethraravindran**
Refactored and cleaned up the code around `end()` methods

### CODE REVIEW

File: Sources/KituraNet/HTTP/HTTPServerResponse.swift [hunk 1]
```diff
@@ -180,11 +156,26 @@ public class HTTPServerResponse: ServerResponse {
...
+        }
+    }
+
+    /// Send response to the client
+    private func sendResponse(channel: Channel, handler: HTTPRequestHandler, status: HTTPResponseStatus, withBody: Bool = false) throws {
+        let response = HTTPResponseHead(version: httpVersion, status: status, headers: headers.httpHeaders())
         channel.write(handler.wrapOutboundOut(.head(response)), promise: nil)
         if withBody && buffer.readableBytes > 0 {
             channel.write(handler.wrapOutboundOut(.body(.byteBuffer(buffer))), promise: nil)
+        } else if buffer.readableBytes > 0 {
```
Comment (pushkarnk) [inline Sources/KituraNet/HTTP/HTTPServerResponse.swift]: This case may not be necessary. We must be able to cover everything in the first `if` case.

File: Sources/KituraNet/HTTP/HTTPServerResponse.swift [hunk 2]
```diff
@@ -180,11 +156,26 @@ public class HTTPServerResponse: ServerResponse {
...
+            do {
+                try self.sendResponse(channel: channel, handler: handler, status: status, withBody: withBody)
+            } catch let error {
+                fatalError("Error: \(error)")
+            }
+        }
+    }
+
+    /// Send response to the client
+    private func sendResponse(channel: Channel, handler: HTTPRequestHandler, status: HTTPResponseStatus, withBody: Bool = false) throws {
```
Comment (pushkarnk) [inline Sources/KituraNet/HTTP/HTTPServerResponse.swift]: `sendResponse` is going to be used for both normal and error responses. So, it makes sense to have `withBody` defaulting to `true`. We can assume that most of the responses do have a body.

Comment (nethraravindran) [reply Sources/KituraNet/HTTP/HTTPServerResponse.swift]: @pushkarnk Thanks! I have addressed the changes

**Comment by pushkarnk** [comment]
Thanks!

**Pull request merged**